packages/services/src/view-service/transaction-planner/index.ts

```diff
@@ -5,11 +5,29 @@ import { Code, ConnectError } from '@connectrpc/connect';
 import { assertSwapAssetsAreNotTheSame } from './assert-swap-assets-are-not-the-same';
 import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
 import { fvkCtx } from '../../ctx/full-viewing-key';
+import { extractAltFee } from '../fees';
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { base64ToUint8Array } from '@penumbra-zone/types/base64';
 
 export const transactionPlanner: Impl['transactionPlanner'] = async (req, ctx) => {
   const services = await ctx.values.get(servicesCtx)();
   const { indexedDb } = await services.getWalletServices();
 
+  // Query IndexedDB directly to check for the existence of spendable 'UM'
+  // notes (native asset balance)
+  const nativeToken = await indexedDb.hasNativeAssetBalance();
+
+  // Initialize the gas fee token using an native staking token's asset ID
+  let gasFeeToken = new AssetId({
+    inner: new Uint8Array(base64ToUint8Array('KeqcLzNx9qSH5+lcJHBB9KNW+YPrBk5dKzvPMiypahA=')),
+  });
```

Comment:
you can get the `stakingAssetId` from the prax registry instead of a string constant

Reply:
modified to 

```
let gasFeeToken = new AssetId({
    inner: stakingTokenId.inner,
});
```
  
  where `stakingTokenId` is retrieved from prax registry.